src/hb-ot-maxp-table.hh

```diff
@@ -73,19 +73,56 @@ struct maxp
     OT::maxp *maxp_prime = (OT::maxp *) hb_blob_get_data (maxp_prime_blob, nullptr);
 
     maxp_prime->set_num_glyphs (plan->gids_to_retain_sorted.len);
+    if (plan->drop_hints)
+      drop_hint_fields (plan, maxp_prime);
 
     bool result = hb_subset_plan_add_table(plan, HB_OT_TAG_maxp, maxp_prime_blob);
     hb_blob_destroy (maxp_prime_blob);
     return result;
   }
 
-  /* We only implement version 0.5 as none of the extra fields in version 1.0 are useful. */
+  static inline void drop_hint_fields (hb_subset_plan_t *plan, OT::maxp *maxp_prime)
+  {
+    if (maxp_prime->version.major == 1)
+    {
+      maxp_prime->version_1_data.maxZones.set (1);
+      maxp_prime->version_1_data.maxTwilightPoints.set (0);
+      maxp_prime->version_1_data.maxStorage.set (0);
+      maxp_prime->version_1_data.maxFunctionDefs.set (0);
+      maxp_prime->version_1_data.maxInstructionDefs.set (0);
+      maxp_prime->version_1_data.maxStackElements.set (0);
+      maxp_prime->version_1_data.maxSizeOfInstructions.set (0);
+    }
+  }
+
   protected:
   FixedVersion<>version;		/* Version of the maxp table (0.5 or 1.0),
 					 * 0x00005000u or 0x00010000u. */
   HBUINT16	numGlyphs;		/* The number of glyphs in the font. */
+
+  struct
```

Comment:
Please move this out to its own struct and add sanitize() and other methods to it, and reference from maxp table. See how it's done in `hb-ot-post-table.hh`. Thanks

Reply:
Done.